src/mesh/nemesis_io_helper.C

```diff
@@ -621,11 +621,14 @@ void Nemesis_IO_Helper::put_node_cmap(std::vector<std::vector<int>> & node_cmap_
 
   for (std::size_t i=0; i<node_cmap_node_ids_in.size(); ++i)
     {
+      int * node_ids_ptr = node_cmap_node_ids_in[i].empty() ?
+        nullptr : &node_cmap_node_ids_in[i][0];
+      int * proc_ids_ptr = node_cmap_proc_ids_in[i].empty() ?
+        nullptr : &node_cmap_proc_ids_in[i][0];
+
       nemesis_err_flag =
-        Nemesis::ne_put_node_cmap(ex_id,
-                                  this->node_cmap_ids[i],
-                                  &node_cmap_node_ids_in[i][0],
-                                  &node_cmap_proc_ids_in[i][0],
+        Nemesis::ne_put_node_cmap(ex_id, this->node_cmap_ids[i],
```

Comment:
Is it definitely safe to call `ne_put_node_cmap` with `nullptr` inputs? It looks like we weren't even guarding against accessing `[0]` of an empty vector before, so this is definitely an improvement.

Reply:
Definitely safe, as in there's a well-documented standard which says we can do this?  No.

Probably safe, as in I've hit a corner case which passes nullptr and our contrib Exodus was okay with that?  Yes.